File: C_C++/random_num.cpp
#include <stdio.h>
#include <stdlib.h>

int readInt(const char* prompt){
    int val;
    int res;
    int c;
    while (1){
        printf("%s", prompt);
        res = scanf("%d", &val);
        if (res == 1){
            break;
        }
        printf("Invalid input, enter numbers only\n");
        while ((c=getchar())!= '\n' && c != EOF);
    }
    return val;
}

int main(){
    int min, max, num;
    char choice;
    
    puts("Random Number Generator");
    min = readInt("Enter the minimum value: ");
    max = readInt("Enter the maximum value: ");

    num = min + rand() % (max - min + 1);
    printf("Your random number is: %d\n", num);
    do{
        
        printf("Would you like to generate another random number? (y/n): ");
        scanf(" %c", &choice);
    
        if(choice == 'y' || choice == 'Y'){
            main();
        } else if(choice == 'n' || choice == 'N'){
            puts("Closing");
        } else {
            puts("Invalid choice, select yes or no");
        }
    } while (choice != 'y' && choice != 'Y' && choice != 'n' && choice != 'N');
    return 0;
}

File: Python/random_num.py
import random

def readInt(prompt):
    while True:
        raw = input(prompt)
        try:
            return int(raw)
        except ValueError:
            print("Invalid input, enter numbers only")

def main():
    print("Random Number Generator")
    min_val = readInt("Enter the minimum value: ")
    max_val = readInt("Enter the maximum value: ")

    num = random.randint(min_val, max_val)
    print(f"Your random number is: {num}")
    while True:
        choice = input("Would you like to generate another random number? (y/n): ")
        if choice in ("y", "Y"):
            main()
            return
        elif choice in ("n", "N"):
            print("Closing")
            return
        else:
            print("Invalid choice, select yes or no")

if __name__ == "__main__":
    main()
